### DISCUSSION TITLE
Needing help: Can't get rnote to run on NixOS?

### DISCUSSION THREAD

**Discussion by TornaxO7** [flxzt/rnote#813] [Development]
Hi! I've switched to `NixOS` and after a harsh landing to get all dependencies for developing on `rnote`, I'm getting the following error message if I'm trying to run `rnote`:

```
(rnote:258888): GLib-GIO-ERROR **: 18:13:23.521: Settings schema 'org.gtk.gtk4.Settings.EmojiChooser' is not installed
[1]    258888 trace trap (core dumped)  rnote
```

# Reproduction steps
1. `PATH` variable (output from `echo $PATH`):
```bash
...:~/.local/bin
```
2. `mkdir -p ~/.local/bin`
2. in `rnote`: `meson setup --prefix=$(readlink -f ~/.local) -Dprofile=devel build`
3. `meson compile -C build`
4. `meson install -C build`
5. `GSETTINGS_SCHEMA_DIR=$(readlink -f ~/.local/share/glib-2.0/schemas) rnote`

May I ask if anyone has an idea what I'm doing wrong?

**Comment by flxzt** [comment]
hey! You didn't post your `shell.nix` file so I can't say if your dev environment is correct, but it might be because you're specifying the `GSETTINGS_SCHEMA_DIR` manually which is probably not the directory where gtk4 has installed the needed settings schema?

**Reply by TornaxO7** [reply]
That's my `shell.nix`:
```nix
let
  rust-overlay = builtins.fetchTarball "https://github.com/oxalica/rust-overlay/archive/master.tar.gz";

  pkgs = import <nixpkgs> {
    overlays = [(import rust-overlay)];
  };

  toolchain = pkgs.rust-bin.fromRustupToolchainFile ./toolchain.toml;
  cairopkg = pkgs.callPackage ./cairo.nix {};
in
  pkgs.mkShell {
    packages = [
      toolchain
      cairopkg
    ];

    inputsFrom = with pkgs; [
      rnote
    ];

    GSETTINGS_SCHEMA_DIR = "/home/tornax/.local/share/glib-2.0/schemas";
  }
```
